File: src/main/java/com/jss/osiris/modules/miscellaneous/service/CustomerOrderOriginService.java
package com.jss.osiris.modules.miscellaneous.service;

import java.util.List;

import com.jss.osiris.modules.miscellaneous.model.CustomerOrderOrigin;

public interface CustomerOrderOriginService {
    public List<CustomerOrderOrigin> getCustomerOrderOrigins();

    public CustomerOrderOrigin getCustomerOrderOrigin(Integer id);

    public CustomerOrderOrigin addOrUpdateCustomerOrderOrigin(CustomerOrderOrigin customerOrderOrigin);

    public List<CustomerOrderOrigin> getByUsername(String username);
}

File: client/src/app/modules/miscellaneous/services/customer.order.origin.service.ts
import { HttpClient, HttpParams } from '@angular/common/http';
import { Injectable } from '@angular/core';
import { AppRestService } from 'src/app/services/appRest.service';
import { CustomerOrderOrigin } from '../../miscellaneous/model/CustomerOrderOrigin';

@Injectable({
  providedIn: 'root'
})
export class CustomerOrderOriginService extends AppRestService<CustomerOrderOrigin>{

  constructor(http: HttpClient) {
    super(http, "miscellaneous");
  }

  getCustomerOrderOrigins() {
    return this.getList(new HttpParams(), "customer-order-origins");
  }
  
   addOrUpdateCustomerOrderOrigin(customerOrderOrigin: CustomerOrderOrigin) {
    return this.addOrUpdate(new HttpParams(), "customer-order-origin", customerOrderOrigin, "Enregistré", "Erreur lors de l'enregistrement");
  }

}
